src/Api/API.php

```diff
@@ -737,41 +737,48 @@ protected function getItem($itemtype, $id, $params = [])
-                                'ON' => [
-                                    'glpi_computers_items'  => 'items_id',
-                                    $connect_table          => 'id'
+            if ($itemtype == "Computer") {
+                $fields['_connections'] = [];
+                foreach ($CFG_GLPI["directconnect_types"] as $connect_type) {
+                    $connect_item = new $connect_type();
+                    if ($connect_item->canView()) {
+                        $connect_table = getTableForItemType($connect_type);
+                        $iterator = $DB->request([
+                            'SELECT'    => [
+                                'glpi_computers_items.id AS assoc_id',
+                                'glpi_computers_items.computers_id AS assoc_computers_id',
+                                'glpi_computers_items.itemtype AS assoc_itemtype',
+                                'glpi_computers_items.items_id AS assoc_items_id',
+                                'glpi_computers_items.is_dynamic AS assoc_is_dynamic',
+                                "$connect_table.*"
+                            ],
+                            'FROM'      => 'glpi_computers_items',
+                            'LEFT JOIN' => [
+                                $connect_table => [
+                                    'ON' => [
+                                        'glpi_computers_items'  => 'items_id',
+                                        $connect_table          => 'id'
+                                    ]
                                 ]
+                            ],
+                            'WHERE'     => [
+                                'computers_id'                      => $id,
+                                'itemtype'                          => $connect_type,
+                                'glpi_computers_items.is_deleted'   => 0
                             ]
-                        ],
-                        'WHERE'     => [
-                            'computers_id'                      => $id,
-                            'itemtype'                          => $connect_type,
-                            'glpi_computers_items.is_deleted'   => 0
-                        ]
-                    ]);
-                    foreach ($iterator as $data) {
-                        $fields['_connections'][$connect_type][] = $data;
+                        ]);
+                        foreach ($iterator as $data) {
+                            $fields['_connections'][$connect_type][] = $data;
+                        }
                     }
                 }
+            } else if ($itemtype == "NetworkPort") {
+                $NetworkportNetworkport = new \NetworkPort_NetworkPort();
+                $otherPortID = $NetworkportNetworkport->getOppositeContact($id);
+                if ($otherPortID) {
+                    $fields['_connections']['NetworkPort'] = $otherPortID;
+                }
```

Comment:
```suggestion
         $networkport_networkport = new \NetworkPort_NetworkPort();
         $opposite_networkport_id = $networkport_networkport->getOppositeContact($id);
         if ($opposite_networkport_id) {
            $fields['_connections']['NetworkPort'][] = [
               "assoc_opposite_networkports_id" => $opposite_networkport_id,
               "assoc_networkports_id" => $id,
               "assoc_id" => $networkport_networkport->fields['id']
            ]
         }
```

Reply:
Add all info from assoc object (```Networkport_Networkport```)

Reply:
All info from assoc object seems really excessive. A user would already have assoc_networkports_id just to run the request, and assoc_id is a mere table index. Do you insist on this change?

Reply:
Moreover, we're returning a NetworkPort, not Networkport_Networkport, so we should just do:
```php
$fields['_connections']['NetworkPort'][] = ["assoc_id" => $opposite_networkport_id];
```